### PR TITLE
Added a class that returns the next rollout user fraction

### PR DESCRIPTION
**Pull request by rocboronat**
In the way to fix the issue #125 , I started creating a class that gives you the next user fraction at Google Play. So, if you say "hey, I'm at 0.05%", it says "ho, the next rollout fraction is "0.1%".

There are my first groovy lines. Feel free to change whatever you need to meet any naming convention you like. By the way, the lines I'm adding to the project are fully tested. No one are calling them yet, but they will be useful for sure.

If you want me to improve something, don't hesitate to share it! :·)

Thanks for your work.

### CODE REVIEW

File: src/main/groovy/de/triplet/gradle/play/PlayRolloutFractionsExtension.groovy [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package de.triplet.gradle.play
+
+class PlayRolloutFractionsExtension {
+
+    def fullRelease = 1.0d;
+
+    def fractions = [0.005d, 0.01d, 0.05d, 0.1d, 0.2d, 0.5d, fullRelease]
```
Comment (ChristianBecker) [inline src/main/groovy/de/triplet/gradle/play/PlayRolloutFractionsExtension.groovy]: We should be consistent about the trailing semicolon

**Comment by ChristianBecker** [comment]
What will happen if you call `getNextUserFraction(0.02d)`? I guess it will return `0.005` in that case. I'm not sure if this is expected behavior.

**Comment by rocboronat** [comment]
Thanks for the feedback, @ChristianBecker !

I just removed all the trailing semicolons, and added a test that does your check and tests that the new method launches an IllegalRolloutException when it recieves an invalid rollout value. I think that it's the best thing that we can do to clarify why we're crashing.

**Comment by rocboronat** [comment]
Any feedback about this PR?

**Comment by rocboronat** [comment]
@bhurling please, consider this PR. It's fully tested and it will be useful for the purpose of the project for sure. In addition, it loves your code standards :·D

**Comment by bhurling** [comment]
This class is nice but apparently it's not used in any task yet. I could think of a task or script that promotes an APK in a staged rollout to the next level. I just wonder if this gradle plugin should really handle these things. Promoting APKs isn't something that is part of the regular build cycle.

cc @ChristianBecker

**Comment by rocboronat** [comment]
Yes @bhurling , this class is not used... 'cos it's new in this PR :·)

By the way, this plugin is wide enough to publish APK's to Google Play, download metadata from Google Play, upload pictures to the app page... so, adding another feature related with publishing to Google Play is good idea.

By the way, how to use the plugin is up to you and your teammates. In our case, we at http://quitnowapp.com upload at 0,5% and go to next level every day. It may be great if the plugin helps us automatizing it, leaving the humans just the task to follow the progressive upload using Fabric.

**Comment by ChristianBecker** [comment]
I'm not sure if we should get this PR merged as-is with code that isn't used in any task yet, as @bhurling already mentioned.

One way to integrate it in a meaningful way could be that we merge this PR with #110 so that we have a task that automatically increases the current `userFraction` if we are on the `promote` track.

This could play nicely with #108 so that we have new `promoteTrack${variationName}` and 
`increaseUserFraction${variationName}` tasks.

Any thoughts, @bhurling, @rocboronat, @shiraji ?

**Comment by rocboronat** [comment]
If it doesn't help anyone, it's better to avoid merging them. By the way, it's easy enough to be retyped at some point of the future... Actually, it was just my hello world in Groovy, and I prefer to share something with the world than just doing a dummy app to try a language :·)

Going to the topic again, totally agree to join this PR with #108 or #110. It will be nice if the task to pass to the next rollout fraction is just `increaseUserFraction`, skipping to say "going to 5% or 10%".

By the way: happy to see that this feature is coming back to topic! 🎉

**Comment by bhurling** [comment]
@rocboronat @ChristianBecker I agree that this feature adds quite some value to the plugin. This should go into one of the next releases. And indeed it should be a combination of all the track/rollout related proposals.

One thing to keep in mind though, is that we now support ABI splits. That means we might have a group of APKs that should be promoted at the same time.

**Comment by SUPERCILEX** [comment]
Hi there, we've moved to Kotlin so this PR is unfortunately obsolete. Sorry about that! 😊 See #104 for discussion.

**Pull request closed without merging**